### PR TITLE
feat(deps): update gts dependency for generate client library

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

File: baselines/bigquery-storage/.eslintrc.json.baseline [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "extends": "./node_modules/gts",
+  "rules": {
```
Comment (JustinBeckwith) [inline baselines/bigquery-storage/.eslintrc.json.baseline]: I don't think we want any of these rules here .... do we?  For each of these, we should rationalize them in `gts`.

File: baselines/bigquery-storage/package.json [hunk 2]
```diff
@@ -30,7 +30,11 @@
     "@types/mocha": "^5.2.7",
     "@types/node": "^12.12.30",
     "c8": "^7.1.0",
-    "gts": "^1.1.2",
+    "gts": "next",
+    "eslint": "^6.8.0",
```
Comment (JustinBeckwith) [inline baselines/bigquery-storage/package.json]: All of the `eslint` dependencies are brought in by `gts` now.  You don't need them here :)

File: templates/typescript_gapic/src/$version/$service_client.ts.njk [hunk 3]
```diff
@@ -298,7 +298,7 @@ export class {{ service.name }}Client {
           if (this._terminated) {
             return Promise.reject('The client has already been closed.');
           }
-          return stub[methodName].apply(stub, args);
+          return stub[methodName](...args);
```
Comment (alexander-fenster) [inline templates/typescript_gapic/src/$version/$service_client.ts.njk]: Auch. I see a mass PR coming :)

**Comment by alexander-fenster** [comment]
Please resolve Justin's comments and please **do not merge** this PR now. Let's postpone it to the time when we are ready to move libraries to `gts@next`. I think I want to have a Docker image with all the changes _before_ this one rolled out first.

File: templates/typescript_gapic/src/$version/$service_client.ts.njk (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply templates/typescript_gapic/src/$version/$service_client.ts.njk]: yes! seems several mass PRs are being planned.

Comment (xiaozhenliu-gg5) [reply templates/typescript_gapic/src/$version/$service_client.ts.njk]: done.

File: baselines/bigquery-storage/package.json (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/package.json]: removed `eslint` related dependencies, thx!

File: baselines/bigquery-storage/.eslintrc.json.baseline (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/.eslintrc.json.baseline]: added extra rules to `gts` and hold this to use the latest gts version.

File: baselines/bigquery-storage/package.json (hunk 2 above)
Comment (JustinBeckwith) [reply baselines/bigquery-storage/package.json]: They're still there though 😆

File: baselines/bigquery-storage/src/v1beta1/big_query_storage_client.ts.baseline [hunk 4]
```diff
@@ -186,6 +186,7 @@ export class BigQueryStorageClient {
         this._opts.fallback ?
           (this._protos as protobuf.Root).lookupService('google.cloud.bigquery.storage.v1beta1.BigQueryStorage') :
           // tslint:disable-next-line no-any
```
Comment (JustinBeckwith) [inline baselines/bigquery-storage/src/v1beta1/big_query_storage_client.ts.baseline]: You can delete the `tslint` specific comments here

File: baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline [hunk 5]
```diff
@@ -19,6 +19,7 @@
 import * as protosTypes from '../protos/protos';
 import * as assert from 'assert';
 import { describe, it } from 'mocha';
+/* eslint-disable @typescript-eslint/no-var-requires */
 const bigquerystorageModule = require('../src');
```
Comment (JustinBeckwith) [inline baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: This is a good time to ask ...  why is this a `require` and not an `import`?

File: baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline [hunk 6]
```diff
@@ -41,7 +42,9 @@ export interface Callback {
 }
 
 export class Operation{
+    /* eslint-disable @typescript-eslint/no-empty-function */
```
Comment (JustinBeckwith) [inline baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: Yaraaaaaaaarr.  This is a silly rule.  Could I trouble you for a PR to disable this rule in `gts`?  I thought I already had ...

File: baselines/disable-packing-test/package.json [hunk 7]
```diff
@@ -30,7 +30,11 @@
     "@types/mocha": "^5.2.7",
     "@types/node": "^12.12.30",
     "c8": "^7.1.0",
-    "gts": "^1.1.2",
+    "gts": "next",
+    "eslint": "^6.8.0",
```
Comment (JustinBeckwith) [inline baselines/disable-packing-test/package.json]: away with ye, eslint

File: baselines/bigquery-storage/package.json (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/package.json]: Oh! I thought it was generator package.json, removed it now from template! :)

File: baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline (hunk 6 above)
Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: is it the same with `"node/no-empty-function": "off",`? in last PR for gts, I set this actually `"node/no-empty-function": "off",` but it complains about empty function. should I explicitely set `"@typescript-eslint/no-empty-function": "off"` instead? Thank you!

File: baselines/bigquery-storage/src/v1beta1/big_query_storage_client.ts.baseline (hunk 4 above)
Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/src/v1beta1/big_query_storage_client.ts.baseline]: removed tslint no-any comment.

File: baselines/disable-packing-test/package.json (hunk 7 above)
Comment (xiaozhenliu-gg5) [reply baselines/disable-packing-test/package.json]: done.

File: baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline (hunk 5 above)
Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: import module will bring the types in :( 
we have many mock methods like `client._innerApiCalls.echo` but in EchoClient `_innerApiCalls` is private, which will throw type checking error.

Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: @alexander-fenster also brought this up in this pr: https://github.com/googleapis/gapic-generator-typescript/pull/354

File: baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline (hunk 6 above)
Comment (JustinBeckwith) [reply baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: Looks like both 😆   Yes please, let's disable this silliness.

Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: Got it, making the PR now, thanks for the catch! Please let me know when new release of gts is coming then I can update this PR :)

Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: PR sent out: https://github.com/google/gts/pull/467

File: baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline (hunk 5 above)
Comment (xiaozhenliu-gg5) [reply baselines/bigquery-storage/test/gapic-big_query_storage-v1beta1.ts.baseline]: spent some time to convert `require` to `import`, it needs more changes than just making `_innerApiCalls ` public. LROperation, gax.CallOptions ... are incompatible with the real types of method calls. Some of them can be forced compatible but lots of `as unknown as someType` will be used.  How about we leave it as it is for now, and as @alexander-fenster suggested, move unit test to `sinon` for removing custom mock codes later as a separate PR.

**Pull request merged**